homeassistant/components/evohome/climate.py

```diff
@@ -385,25 +385,15 @@ def preset_mode(self) -> str | None:
         """Return the current preset mode, e.g., home, away, temp."""
         return TCS_PRESET_TO_HA.get(self._evo_tcs.systemModeStatus["mode"])
 
-    @property
-    def min_temp(self) -> float:
-        """Return None as Controllers don't have a target temperature."""
-        return None
-
-    @property
-    def max_temp(self) -> float:
-        """Return None as Controllers don't have a target temperature."""
-        return None
-
     async def async_set_temperature(self, **kwargs) -> None:
         """Raise exception as Controllers don't have a target temperature."""
         raise NotImplementedError("Evohome Controllers don't have target temperatures.")
 
-    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
         """Set an operating mode for a Controller."""
-        await self._set_tcs_mode(HA_HVAC_TO_TCS.get(hvac_mode))
+        await self._set_tcs_mode(HA_HVAC_TO_TCS[hvac_mode])
```

Comment:
I'd first check if hvac_mode is in the conversion dict. There's currently nothing validating that the hvac mode is in the entity supported hvac modes in the climate service handler. So we'll need to check that here.